test/src/unit-capi-consolidation.cc

```diff
@@ -5865,4 +6020,337 @@ TEST_CASE_METHOD(
+    // Consolidation works.
+    write_sparse_unordered();
+    write_sparse_full();
+    consolidate_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(1, 2, 0);
+
+    // Vacuum works.
+    vacuum_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(1, 0, 0);
+
+    // Second consolidation works.
+    consolidate_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(2, 0, 0);
+
+    // Second vacuum works.
+    vacuum_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(1, 0, 0);
+
+    // After fragment consolidation and vacuuming, array is still valid.
+    consolidate_sparse();
+    vacuum_sparse();
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(1, 1, 1);
+
+    // Consolidation to get rid of ignore file.
+    consolidate_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(2, 1, 1);
+
+    // Second vacuum works.
+    vacuum_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(1, 0, 0);
+  }
+
+  SECTION("- write (encrypted) unordered, full") {
+    remove_sparse_array();
+    encryption_type_ = TILEDB_AES_256_GCM;
+    encryption_key_ = "0123456789abcdeF0123456789abcdeF";
+    create_sparse_array();
+
+    // Consolidation works.
+    write_sparse_unordered();
+    write_sparse_full();
+    consolidate_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(1, 2, 0);
+
+    // Vacuum works.
+    vacuum_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(1, 0, 0);
+
+    // Second consolidation works.
+    consolidate_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(2, 0, 0);
+
+    // Second vacuum works.
+    vacuum_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(1, 0, 0);
+
+    // After fragment consolidation and vacuuming, array is still valid.
+    consolidate_sparse();
+    vacuum_sparse();
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(1, 1, 1);
+
+    // Consolidation to get rid of ignore file.
+    consolidate_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(2, 1, 1);
+
+    // Second vacuum works.
+    vacuum_sparse("commits");
+    read_sparse_unordered_full();
+    check_commits_dir_sparse(1, 0, 0);
+  }
+
+  remove_sparse_array();
+}
+
+#ifndef _WIN32
+TEST_CASE_METHOD(
+    ConsolidationFx,
+    "C API: Test consolidation, sparse, commits, mixed versions",
+    "[capi][consolidation][commits][mixed-versions]") {
+  remove_sparse_array();
+
+  // Get the v11 sparse array.
+  std::string v11_arrays_dir =
+      std::string(TILEDB_TEST_INPUTS_DIR) + "/arrays/sparse_array_v11";
+  REQUIRE(
+      tiledb_vfs_copy_dir(
+          ctx_, vfs_, v11_arrays_dir.c_str(), SPARSE_ARRAY_NAME) == TILEDB_OK);
+
+  // Write v11 fragment.
```

Comment:
@KiterLuc double-checkin: is this function is writing a v11 fragment or writing a _new_ fragment at the current version, to the v11 array copied above? cc @bekadavis9